tests/unittests/test_command_line_parser.py

```diff
@@ -0,0 +1,653 @@
+    key_aliases = ["log-level"]
+    params = {
+        'key_aliases': key_aliases,
+        'delimiter': '='
+    }
+    val = command_line_parser.parse_cmdline(params=params, chained=command_line)
+    log.debug("return value is {0}".format(val))
+    expected_value = (True, ["debug"])
+    assert val == expected_value
+
+
+def test_key_alias_not_found_in_cmdline():
+    """
+        Key alias is not found in the commandLine
+        Expected Status : True
+    """
+    log.info("Executing test_key_alias_not_found_in_cmdline")
+    command_line = {"cmdline":"dockerd --config-file=\"/etc/docker/daemon.json\" --log-level=\"debug\""}
+    key_aliases = ["not_found"]
+    params = {
+        'key_aliases': key_aliases,
+        'delimiter': '='
+    }
+    val = command_line_parser.parse_cmdline(params=params, chained=command_line)
+    log.debug("return value is {0}".format(val))
+    expected_value = (True, [])
+    assert val == expected_value
+
+
+def test_multiple_return_values():
+    """
+        commandline has multiple values corresponding to the key
+        Expected Status : True
+    """
+    log.info("Executing test_multiple_return_values")
+    command_line = {"cmdline" : "docker run -v=\"a:b\" --volume=\"d:e\" --log-level=\"debug\""}
+    key_aliases = ["v", "volume"]
+    params = {
+              'key_aliases': key_aliases,
+              'delimiter': '='
+              }
+    val = command_line_parser.parse_cmdline(params = params, chained=command_line)
+    log.debug("return value is {0}".format(val))
+    expected_value = (True, ["a:b", "d:e"])
+    assert val == expected_value
+
+
+def test_multiple_key_aliases():
+    """
+        result has results for multiple keys
+        Expected Status : True
+    """
+    log.info("Executing test_multiple_key_aliases")
+    command_line = {"cmdline":"docker run -v=\"a:b\" -v=\"d:e\" --log-level=\"debug\""}
+    key_aliases = ["volume", "v"]
+    params = {
+        'key_aliases': key_aliases,
+        'delimiter': '='
+    }
+    val = command_line_parser.parse_cmdline(params=params, chained=command_line)
+    log.debug("return value is {0}".format(val))
+    expected_value = (True, ["a:b", "d:e"])
+    assert val == expected_value
+
+
+def test_values_with_single_quotes():
+    """
+        values in command line have single-quotes
+        Expected Status : True
+    """
+    log.info("Executing test_values_with_single_quotes")
+    command_line = {"cmdline":"docker run -v=\'a:b\' -v=\'d:e\' --log-level=\'debug\'"}
+    key_aliases = ["volume", "v"]
+    params = {
+        'key_aliases': key_aliases,
+        'delimiter': '='
+    }
+    val = command_line_parser.parse_cmdline(params=params, chained=command_line)
+    log.debug("return value is {0}".format(val))
+    expected_value = (True, ["a:b", "d:e"])
+    assert val[0] == expected_value[0]
+    assert val[1] == expected_value[1]
+
+
+def test_second_regex():
```

Comment:
Please add a testcase in case there are multiple spaces in between alias and value

Reply:
There is already such a test case `test_extra_spaces_different_delimiter`